python/ray/_private/logging/structured_logging.py

```diff
@@ -0,0 +1,46 @@
+import logging
+import ray
+import json
+
+from ray._private.ray_constants import (
+    LOGGER_RAY_ATTRS,
+    LOGRECORD_STANDARD_ATTRS,
+)
+
+
+class CoreContextFilter(logging.Filter):
+    def filter(self, record):
+        runtime_context = ray.get_runtime_context()
+        record.job_id = runtime_context.get_job_id()
+        record.worker_id = runtime_context.get_worker_id()
+        record.node_id = runtime_context.get_node_id()
+        if runtime_context.worker.mode == ray.WORKER_MODE:
+            actor_id = runtime_context.get_actor_id()
+            if actor_id is not None:
+                record.actor_id = actor_id
+            task_id = runtime_context.get_task_id()
+            if task_id is not None:
+                record.task_id = task_id
+        return True
+
+
+class JSONFormatter(logging.Formatter):
+    def format(self, record):
+        record_format = {
+            "ts": self.formatTime(record),
```

Comment:
The default time format we want to use is `%Y-%m-%d %H:%M:%S,%e%z`

Reply:
TLDR: I would recommend using the default format plus a new field, `timezone`, in the JSON.

`%e` is the day of date. For example, `%e` of May. 3 is 3. I guess you are talking about milli second. However, there is no directive for milli second. Based on this [Python doc](https://docs.python.org/3/library/time.html#time.strftime), it only has `%f` which is micro seconds. In addition, the doc also mentions that "The `%f` format directive only applies to [strptime()](https://docs.python.org/3/library/time.html#time.strptime), not to [strftime()](https://docs.python.org/3/library/time.html#time.strftime)".

The function `formatTime` calls `strftime` under the hood. See the [source code](https://github.com/python/cpython/blob/2770d5caca42d48102f8e18210132a964c34af7c/Lib/logging/__init__.py#L627-L652) for more details. You can also see how it generates milli second, which requires additional efforts to create.

We can build the timestamp ourselves, but we can't use the utility function provided by the logging module, which I think is not a good practice.

Reply:
Sync with @jjyao offline. Currently, we have decided to use the default format plus a new field timezone.